server/test/unit/paTest/handleMessageClientsTest.js

```diff
@@ -0,0 +1,226 @@
+const testBadFromPhoneNumber = '+17781111111'
+const testTwilioMessage = 'Hello, Brave!'
+
+const expectedArguments = [
+  ...successfulPhoneNumbers.map(phoneNumber => {
+    return { toPhoneNumber: phoneNumber, fromPhoneNumber: testFromPhoneNumber, twilioMessage: testTwilioMessage }
+  }),
+  ...failPhoneNumbers.map(phoneNumber => {
+    return { toPhoneNumber: phoneNumber, fromPhoneNumber: testBadFromPhoneNumber, twilioMessage: testTwilioMessage }
+  }),
+]
+
+// handleMessageClients needs:
+// - id
+// - displayName
+// - responderPhoneNumbers
+// - fallbackPhoneNumbers
+// - heartbeatPhoneNumbers
+// - fromPhoneNumber
+// from each client
+const activeClients = [
+  factories.clientFactory({
+    id: 'client-just-responder-phone-number',
+    displayName: 'Client Just Responder Phone Number',
+    responderPhoneNumbers: [successfulPhoneNumbers[0]],
+    fallbackPhoneNumbers: [],
+    heartbeatPhoneNumbers: [],
+    fromPhoneNumber: testFromPhoneNumber,
+  }),
+  factories.clientFactory({
+    id: 'client-just-fallback-phone-number',
+    displayName: 'Client Just Fallback Phone Number',
+    responderPhoneNumbers: [],
+    fallbackPhoneNumbers: [successfulPhoneNumbers[1]],
+    heartbeatPhoneNumbers: [],
+    fromPhoneNumber: testFromPhoneNumber,
+  }),
+  factories.clientFactory({
+    id: 'client-just-heartbeat-phone-number',
+    displayName: 'Client Just Heartbeat Phone Number',
+    responderPhoneNumbers: [],
+    fallbackPhoneNumbers: [],
+    heartbeatPhoneNumbers: [successfulPhoneNumbers[2]],
+    fromPhoneNumber: testFromPhoneNumber,
+  }),
+  factories.clientFactory({
+    id: 'client-duplicate-phone-numbers',
+    displayName: 'Client Duplicate Phone Numbers',
+    responderPhoneNumbers: [successfulPhoneNumbers[3]],
+    fallbackPhoneNumbers: [successfulPhoneNumbers[3]],
+    heartbeatPhoneNumbers: [successfulPhoneNumbers[3]],
+    fromPhoneNumber: testFromPhoneNumber,
+  }),
+  factories.clientFactory({
+    id: 'client-no-duplicate-phone-numbers',
+    displayName: 'Client No Duplicate Phone Numbers',
+    responderPhoneNumbers: [successfulPhoneNumbers[4]],
+    fallbackPhoneNumbers: [successfulPhoneNumbers[5]],
+    heartbeatPhoneNumbers: [successfulPhoneNumbers[6]],
+    fromPhoneNumber: testFromPhoneNumber,
+  }),
+  factories.clientFactory({
+    id: 'client-duplicate-and-multiple-phone-numbers',
+    displayName: 'Client Duplicate And Multiple Phone Numbers',
+    responderPhoneNumbers: [successfulPhoneNumbers[7]],
+    fallbackPhoneNumbers: [successfulPhoneNumbers[8], successfulPhoneNumbers[9]],
+    heartbeatPhoneNumbers: [successfulPhoneNumbers[7]],
+    fromPhoneNumber: testFromPhoneNumber,
+  }),
+  factories.clientFactory({
+    id: 'client-bad-from-phone-number',
+    displayName: 'Client Bad From Phone Number',
+    responderPhoneNumbers: [failPhoneNumbers[0]],
+    fallbackPhoneNumbers: [],
+    heartbeatPhoneNumbers: [],
+    fromPhoneNumber: testBadFromPhoneNumber,
+  }),
```

Comment:
I'm glad you included a client with a phone number that fails, among the successful active clients.